samples/websocket-gateway/Gateway_2/WebSocketGateway/WebSocketChannelSender.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading.Tasks;
+using NServiceBus.Gateway;
+using WebSocketSharp;
+
+#region WebScoketGateway-ChannelSender
+public class WebSocketChannelSender :
+    IChannelSender
+{
+    public Task Send(string remoteAddress, IDictionary<string, string> headers, Stream data)
+    {
+        using (var webSocket = new WebSocket(remoteAddress))
+        {
+            webSocket.Connect();
+            var bytes = GetBytes(headers, data);
+            webSocket.Send(bytes);
```

Comment:
Possibly. The SendAsync method doesn't return a Task though. You have to pass in a continuation. I figured this would just add unnecessary complexity to the sample